src/cc/bcc_version.h.in

```diff
@@ -1,6 +1,12 @@
 #ifndef LIBBCC_VERSION_H
 #define LIBBCC_VERSION_H
 
-#define LIBBCC_VERSION "@REVISION@"
+#define LIBBCC_VERSION_STRING "@REVISION@"
+#define LIBBCC_MAJOR_VERSION @REVISION_MAJOR@
+#define LIBBCC_MINOR_VERSION @REVISION_MINOR@
+#define LIBBCC_PATCH_VERSION @REVISION_PATCH@
+
+#define LIBBCC_VERSION(a, b, c) (((a) << 16) + ((b) << 8) + ((c) > 255 ? 255 : (c)))
```

Comment:
I should probably not reuse LIBBCC_VERSION here and rather create a new macro. First it does not seem easy to check if something is defined and use LIBBCC_VERSION in the context of using it to compare the version number. See error in https://github.com/chantra/gobpf/commit/c9487712aa78e0a0f1b4d72e016d6dbad0059dcc#diff-84bebc733df1ff5a43002386008140ff785cd0af93d50eb1257dbaa3a2421eacR42

where 
```
#if defined(LIBBCC_VERSION_CODE) && LIBBCC_VERSION_CODE > LIBBCC_VERSION(0, 24, 0)
```
Would fail with

```
error: token ""0.20.0-14278bf1"" is not valid in preprocessor expressions
```

Second. There is also some minimal use of current LIBBCC_VERSION out there: https://github.com/search?q=LIBBCC_VERSION&type=code


So probably better to use another name.

Reply:
Maybe just use __LIBBCC_VERSION macro to imply it is a internal macro?

Reply:
Works for me. I will get back to this when back from vacation :)

Reply:
> Maybe just use __LIBBCC_VERSION macro to imply it is a internal macro to assemble the version code?

something also needs to be “publicly” visible though. This way API users can compare which BCC version they are running against, similar to https://android.googlesource.com/kernel/common/+/bcmdhd-3.10/drivers/net/wireless/bcmdhd/include/linuxver.h#33

It is more of a naming problem than technical one :)


What I see used across different project is a macro allowing to check that the version is greater or equal to x.
So maybe we can strike a compromise by having an “internal” macro to compute the code, and a public one to check if greater or equal. Something  like:
```
#define __LIBBCC_VERSION(a, b, c) (((a) << 16) + ((b) << 8) + ((c) > 255 ? 255 : (c)))
#define LIBBCC_VERSION_GEQ(a,b,c) LIBBCC_VERSION_CODE >= __LIBBCC_VERSION(a, b, c)
```

Reply:
I don't have special preference. What you proposed works for me too.